lib/ansible/modules/remote_management/oneview/oneview_storage_system.py

```diff
@@ -0,0 +1,233 @@
+'''
+
+EXAMPLES = '''
+- name: Add a Storage System with one managed pool (before API500)
+  oneview_storage_system:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 300
+    state: present
+    data:
+        credentials:
+            ip_hostname: 172.18.11.12
+            username: username
+            password: password
+        managedDomain: TestDomain
+        managedPools:
+          - domain: TestDomain
+            type: StoragePoolV2
+            name: CPG_FC-AO
+            deviceType: FC
+
+  no_log: true
+  delegate_to: localhost
+
+- name: Add a StoreServ Storage System with one managed pool (API500 onwards)
+  oneview_storage_system:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: present
+    data:
+      credentials:
+          username: username
+          password: password
+      hostname: 172.18.11.12
+      family: StoreServ
+      deviceSpecificAttributes:
+          managedDomain: TestDomain
+          managedPools:
+              - domain: TestDomain
+                name: CPG_FC-AO
+                deviceType: FC
+
+  no_log: true
+  delegate_to: localhost
+
+- name: Remove the storage system by its IP (before API500)
+  oneview_storage_system:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 300
+    state: absent
+    data:
+        credentials:
+            ip_hostname: 172.18.11.12
+
+  no_log: true
+  delegate_to: localhost
+
+- name: Remove the storage system by its IP (API500 onwards)
+  oneview_storage_system:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 500
+    state: absent
+    data:
+        credentials:
+            hostname: 172.18.11.12
+
+  no_log: true
+  delegate_to: localhost
+'''
+
+RETURN = '''
+storage_system:
+    description: Has the OneView facts about the Storage System.
+    returned: On state 'present'. Can be null.
+    type: dict
+'''
+
+from ansible.module_utils.six.moves import filter
+from ansible.module_utils.oneview import OneViewModuleBase, OneViewModuleValueError
+
+
+class StorageSystemModule(OneViewModuleBase):
+    MSG_ADDED = 'Storage System added successfully.'
+    MSG_UPDATED = 'Storage System updated successfully.'
+    MSG_ALREADY_PRESENT = 'Storage System is already present.'
+    MSG_DELETED = 'Storage System deleted successfully.'
+    MSG_ALREADY_ABSENT = 'Storage System is already absent.'
+    MSG_MANDATORY_FIELDS_MISSING = "At least one mandatory field must be provided: name or credentials.hostname" \
+                                   "(credentials.ip_hostname if API version lower than 500 )."
+    MSG_CREDENTIALS_MANDATORY = "The attribute 'credentials' is mandatory for Storage System creation."
+
+    def __init__(self):
+        argument_spec = dict(
+            state=dict(
+                required=True,
```

Comment:
Should be `default='present'` and this whole `state` should fit into one line.
I also suggest reordering the argument spec as mentioned in [here](https://github.com/ansible/ansible/pull/32798#discussion_r150491001)